File: Data Structures & Algorithms/top-k-elements-in-list/submission-5.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        std::unordered_map<int, int> count;
        for (int num : nums)  {
           count[num]++;
        }

        std::priority_queue<std::pair<int, int>, vector<pair<int, int>>, greater<pair<int, int>>> heap;
        for (auto& entry : count) {
            heap.push({entry.second, entry.first});
            if (heap.size() > k) {
                heap.pop();
            }
        }

        vector<int> result;

        while (!heap.empty()) {
            result.push_back(heap.top().second);
            heap.pop();
        }
        return result;
    }
};

File: Data Structures & Algorithms/validate-parentheses/submission-5.py

class Solution:
    def isValid(self, string: str) -> bool:
        stack = []

        for s in string:
            if s == ')' and (len(stack) == 0 or stack.pop() != '('):
                return False
            elif s == '}' and (len(stack) == 0 or stack.pop() != '{'):
                return False
            elif s == ']' and (len(stack) == 0 or stack.pop() != '['):
                return False
            elif s in ['(', '[', '{']:
                stack.append(s)
        return len(stack) == 0